### ISSUE TITLE
BREAKING: Change .destroy(onclose) to .destroy(err)

### ISSUE THREAD

**Issue by dex4er** [feross/simple-websocket#27]
This module overrides `destroy` method in incompatible way with `Duplex.destroy` from Node 8.x

https://nodejs.org/api/stream.html#stream_writable_destroy_error

so `destroy` method doesn't support any callback and its only argument is an `Error` object.

It would be nice to provide `Socket` class 100% compatible with `Duplex` class and call `super.destroy()` if an argument is an `Error` and do not override `_destroy` method as far as is implemented internally by `Duplex` class.

**Comment by dex4er** [comment]
BTW, these are typings for TypeScript: https://github.com/dex4er/DefinitelyTyped/tree/simple-websocket/types/simple-websocket

Because of this incompatibility I couldn't use current simple-websocket in my TypeScript project :( 
Alternative is to not provide `destroy` in typings and the do not use it at all.

**Comment by feross** [comment]
Unfortunately, `readable-stream` hasn't implemented the new `destroy()` behavior yet. Here's the issue for it: https://github.com/nodejs/readable-stream/issues/283

**Comment by feross** [comment]
Alright, we can support the new API even though readable-stream doesn't support it yet. I just had to include a little shim for destroy().

To update your code, change this:

```js
socket.destroy(callback)
```

To this instead:

```js
socket.on('close', callback)
socket.destroy()
```

**Comment by feross** [comment]
Released as 7.0.0.
